src/providers/Preferences/UserPreferences.provider.jsx

```diff
@@ -0,0 +1,81 @@
+import React, { useReducer, useEffect, createContext, useContext } from 'react';
+
+import { lightTheme, darkTheme } from '../../utils/theme';
+import { userPreferencesReducer, initialState } from './UserPreferences.reducer';
+import { addFavorite, doInvertTheme, removeFavorite } from './UserPreferences.actions';
+import { storage } from '../../utils/storage';
+import { USER_PREFERENCES_STORAGE_KEY } from '../../utils/constants';
+import { useAuth } from '../Auth';
+
+const UserPreferencesContext = createContext(null);
+
+function useUserPreferences() {
+  const context = useContext(UserPreferencesContext);
+  if (!context) {
+    throw new Error(`Can't use "useAuth" without an AuthProvider!`);
+  }
+  return context;
+}
+
+function getUserStorageKey(user) {
+  return user ? `${USER_PREFERENCES_STORAGE_KEY}-${user.id}` : '';
+}
+
```

Comment:
Great use of `useReducer` + `useContext`, I like the way you define the actions, the reducer and the context in their own files